pkg/registry/core/service/ipallocator/controller/repairip.go

```diff
@@ -111,38 +126,99 @@ func NewRepairIPAddress(interval time.Duration,
 	}
 
 	r := &RepairIPAddress{
-		interval:        interval,
-		client:          client,
-		networkByFamily: networkByFamily,
-		serviceLister:   serviceInformer.Lister(),
-		servicesSynced:  serviceInformer.Informer().HasSynced,
-		ipAddressLister: ipAddressInformer.Lister(),
-		ipAddressSynced: ipAddressInformer.Informer().HasSynced,
-		broadcaster:     eventBroadcaster,
-		recorder:        recorder,
-		clock:           clock.RealClock{},
+		interval:         interval,
+		client:           client,
+		networkByFamily:  networkByFamily,
+		serviceLister:    serviceInformer.Lister(),
+		servicesSynced:   serviceInformer.Informer().HasSynced,
+		ipAddressLister:  ipAddressInformer.Lister(),
+		ipAddressSynced:  ipAddressInformer.Informer().HasSynced,
+		svcQueue:         workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "services"),
+		ipQueue:          workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "ipaddresses"),
+		workerLoopPeriod: time.Second,
+		broadcaster:      eventBroadcaster,
+		recorder:         recorder,
+		clock:            clock.RealClock{},
 	}
 
+	serviceInformer.Informer().AddEventHandlerWithResyncPeriod(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			key, err := cache.MetaNamespaceKeyFunc(obj)
+			if err == nil {
+				r.svcQueue.Add(key)
+			}
+		},
+		UpdateFunc: func(old interface{}, new interface{}) {
+			key, err := cache.MetaNamespaceKeyFunc(new)
+			if err == nil {
+				r.svcQueue.Add(key)
+			}
+		},
+		DeleteFunc: func(obj interface{}) {
+			// IndexerInformer uses a delta queue, therefore for deletes we have to use this
+			// key function.
+			key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
+			if err == nil {
+				r.svcQueue.Add(key)
+			}
+		},
+	}, interval)
+
+	ipAddressInformer.Informer().AddEventHandlerWithResyncPeriod(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			key, err := cache.MetaNamespaceKeyFunc(obj)
+			if err == nil {
+				r.ipQueue.Add(key)
+			}
+		},
+		UpdateFunc: func(old interface{}, new interface{}) {
+			key, err := cache.MetaNamespaceKeyFunc(new)
+			if err == nil {
+				r.ipQueue.Add(key)
+			}
+		},
+		DeleteFunc: func(obj interface{}) {
+			// IndexerInformer uses a delta queue, therefore for deletes we have to use this
+			// key function.
+			key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
+			if err == nil {
+				r.ipQueue.Add(key)
+			}
+		},
+	}, interval)
```

Comment:
This generates a fake update event every interval so all cache elements gets processed again, this is similar to previous repair loop, but I don't have clear we need it here